src/nwb_conversion_tools/datainterfaces/ecephys/spikeglx/spikeglxdatainterface.py

```diff
@@ -107,6 +123,36 @@ def get_conversion_options(self):
         conversion_options = dict(write_as="raw", es_key="ElectricalSeries_raw", stub_test=False)
         return conversion_options
 
+    def get_device_metadata(self) -> dict:
+        """Returns a device with description including the metadat as described here
+        # https://billkarsh.github.io/SpikeGLX/Sgl_help/Metadata_30.html
+
+        Returns
+        -------
+        dict
+            a dict containing the metadata necessary for creating the device
+        """
+
+        meta = self.meta
+        metadata_dict = dict()
+        if "imDatPrb_type" in self.meta:
+            probe_type_to_probe_description = {"0": "NP1.0", "21": "NP2.0(1-shank)", "24": "NP2.0(4-shank)"}
+            probe_type = str(meta["imDatPrb_type"])
+            probe_type_description = probe_type_to_probe_description[probe_type]
+            metadata_dict.update(probe_type=probe_type, probe_type_description=probe_type_description)
+
+        if "imDatFx_pn" in self.meta:
+            metadata_dict.update(flex_part_number=meta["imDatFx_pn"])
+
+        if "imDatBsc_pn" in self.meta:
+            metadata_dict.update(connected_base_station_part_number=meta["imDatBsc_pn"])
+
+        metadata_json = json.dumps(metadata_dict)
+        description_string = f"Imec device - metadata = {metadata_json}"
```

Comment:
```suggestion
        description_string = metadata_json
```

would you mind just having this entire field be valid json?

Reply:
Yes.

There is the following edge case though. when no fields are found, the line as you suggest would return description="{}", what do you think would be good to have in that case?

Reply:
I don't know, in that case "no description" I guess

Reply:
All right, this is done.